python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -80,12 +88,29 @@ def __init__(
         # Deprecate: Raise in 2.6, remove in 2.7
         sync_config: Optional[SyncConfig] = None,
     ):
+        self._local_experiment_path: str = None
+        self._remote_experiment_path: Optional[str] = None
+
+        # If the user passes in a remote checkpoint path,
+        # Set the remote experiment path to this path, and set
+        # the local experiment path to a temp directory.
+        if not is_local_path(experiment_checkpoint_path):
+            self._remote_experiment_path = experiment_checkpoint_path
+
+            # Create a temp directory to store downloaded checkpoint files if
+            # they are pulled from a remote `experiment_checkpoint_path`.
+            self._local_experiment_path = tempfile.TemporaryDirectory(
```

Comment:
Should we make this direcotry canonical? e.g. via `uuid5`

Reply:
I believe the `TemporaryDirectory` will already append a unique string after the prefix!

Reply:
Oh what I mean is do we want to always have the same directory, so we don't download multiple times if we restore multiple times.

The main reason for this is that this will download the full experiment dir (including checkpoints) which may be expensive

Reply:
I guess this is more of an optimization, so also happy to go as is

Reply:
Oh, got it. This will only download a few files from
the experiment (experiment state, results csv/json, params.pkl) -- no checkpoints or artifacts. So, I thought it was lightweight enough.

A few ideas I had for a follow-up:
- Cleanup the temp dir on experiment analysis destructor.
- Move result grid creation to get_results so that it doesn't always happen on restore.